### PR TITLE
Added or improved documentation to a few more String methods

### PR DESCRIPTION
**Pull request by novaplusplus**
Includes bigrams(), chr(), similarity(), is_valid_int(), is_valid_float(), is_valid_identifier()

### CODE REVIEW

File: doc/classes/String.xml [hunk 1]
```diff
@@ -328,7 +336,14 @@
...
 			<return type="bool" />
 			<description>
-				Returns [code]true[/code] if this string contains a valid float.
+				Returns [code]true[/code] if this string contains a valid float. This is inclusive of integers, and also supports exponents:
+				[codeblock]
+				print("1.7".is_valid_float()) # Prints "true"
+				print("24".is_valid_float()) # Prints "true"
+				print("7e3".is_valid_float()) # Prints "true"
+				print("24".is_valid_float()) # Prints "true"
+				print("B".is_valid_float()) # Prints "false"
```
Comment (aaronfranke) [inline doc/classes/String.xml:345]: For the sake of readability, let's choose text that doesn't look like the number 8.

```suggestion
				print("Hello".is_valid_float()) # Prints "false"
```

Comment (novaplusplus) [reply doc/classes/String.xml:345]: Good catch, I'll get to it shortly.

Comment (novaplusplus) [reply doc/classes/String.xml]: Ok, done.

**Review by aaronfranke — APPROVED** [review]
Looks good to me.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.4.

**Comment by novaplusplus** [comment]
It took me this long to realize that the line `print("24".is_valid_float()) # Prints "true"` is repeated twice...

**Pull request merged**